src/controllers/posts/getPostsCohortController.ts

```diff
@@ -0,0 +1,43 @@
+import { Response, NextFunction } from 'express';
+import { CustomError } from '../../utils';
+import { getAllPostsCohortQuery } from '../../database';
+import { GetCohortPostsRequest } from '../../utils/types';
+
+const getPostsCohortController = async (
+  req: GetCohortPostsRequest,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { page } = req.query;
+    const offset = (Number(page || 1) - 1) * 10;
+    const { stringId } = req.params;
+    const id = +stringId;
+    const dataUser = req.user;
+    const roles = dataUser?.roles;
+
+    if (!roles || dataUser.isAdmin === true) {
+      const { rows, rowCount } = await getAllPostsCohortQuery({ id, offset });
+      if (!rowCount) throw new CustomError('The cohort does not exist', 404);
+
+      return res.status(200).json({ message: 'Success', data: rows });
+    }
+
+    const findCohortAccess = roles.filter((role) => role.cohort_id === id);
+    if (!findCohortAccess.length) throw new CustomError('Unauthorized', 400);
+    const { rows, rowCount } = await getAllPostsCohortQuery({ id, offset });
+    if (!rowCount) throw new CustomError('The cohort does not exist', 404);
+
+    return res.status(200).json({
+      error: 'false',
+      data: {
+        message: 'Success',
+        posts: rows,
+      },
+    });
```

Comment:
I have some notes for you to improve your code:
- Replace the `filter` method with the `some` method when checking for cohort access. The `some` method stops iterating over the array once a matching role is found, making it more efficient.
- Changed `return res.status(200).json(...)` to `res.status(200).json(...)`. The `return` statement is not needed in this case.
- Removed the `return next(err)` statement and replaced it with `next(err)`. Since it's the last statement in the function, the `return` is not necessary.

here is e.g: 
```ts
const { roles, isAdmin } = req.user;

if (!isAdmin && (roles === undefined || !roles.some((role) => role.cohort_id === id))) {
    throw new CustomError('Unauthorized', 400);
}

const { rows, rowCount } = await getAllPostsCohortQuery({ id, offset });

if (rowCount === 0) {
  return res.status(200).json({ message: 'No content' });
}

const responseData = {
  message: 'Success',
  data: {
    posts: rows,
  },
};

res.status(200).json(responseData);
```